python/ql/lib/semmle/python/dataflow/new/internal/FlowSummaryImplSpecific.qll

```diff
@@ -0,0 +1,191 @@
+/**
+ * Provides Python specific classes and predicates for defining flow summaries.
+ */
+
+private import python
+private import DataFlowPrivate
+private import DataFlowPublic
+private import DataFlowImplCommon
+private import FlowSummaryImpl::Private
+private import FlowSummaryImpl::Public
+private import semmle.python.dataflow.new.FlowSummary as FlowSummary
+
+class SummarizedCallableBase = string;
+
+DataFlowCallable inject(SummarizedCallable c) { result.asLibraryCallable() = c }
+
+/** Gets the parameter position of the instance parameter. */
+ArgumentPosition instanceParameterPosition() { none() } // disables implicit summary flow to `this` for callbacks
+
+/** Gets the synthesized summary data-flow node for the given values. */
+Node summaryNode(SummarizedCallable c, SummaryNodeState state) { result = TSummaryNode(c, state) }
+
+/** Gets the synthesized data-flow call for `receiver`. */
+SummaryCall summaryDataFlowCall(Node receiver) { receiver = result.getReceiver() }
+
+/** Gets the type of content `c`. */
+DataFlowType getContentType(Content c) { any() }
+
+/** Gets the return type of kind `rk` for callable `c`. */
+bindingset[c, rk]
+DataFlowType getReturnType(SummarizedCallable c, ReturnKind rk) { any() }
+
+/**
+ * Gets the type of the `i`th parameter in a synthesized call that targets a
+ * callback of type `t`.
+ */
+bindingset[t, i]
+DataFlowType getCallbackParameterType(DataFlowType t, int i) { any() }
+
+/**
+ * Gets the return type of kind `rk` in a synthesized call that targets a
+ * callback of type `t`.
+ */
+DataFlowType getCallbackReturnType(DataFlowType t, ReturnKind rk) { any() }
+
+/**
+ * Holds if an external flow summary exists for `c` with input specification
+ * `input`, output specification `output`, kind `kind`, and a flag `generated`
+ * stating whether the summary is autogenerated.
+ */
+predicate summaryElement(
+  FlowSummary::SummarizedCallable c, string input, string output, string kind, boolean generated
+) {
+  exists(boolean preservesValue |
+    c.propagatesFlowExt(input, output, preservesValue) and
+    (if preservesValue = true then kind = "value" else kind = "taint") and
+    generated = false
+  )
+}
+
+/**
+ * Gets the summary component for specification component `c`, if any.
+ *
+ * This covers all the Python-specific components of a flow summary.
+ */
+SummaryComponent interpretComponentSpecific(AccessPathToken c) {
+  c = "Argument[_]" and // Ruby has this
```

Comment:
I don't understand the inline comment about Ruby. Should this be deleted? if not, I'd like to understand what it actually means :sweat_smile:

Reply:
I think I just kept this as an example, I have deleted it now and the tests still pass :-)